experimenter/experimenter/jetstream/tasks.py

```diff
@@ -23,13 +23,18 @@ def fetch_experiment_data(experiment_id):
     experiment = None
     try:
         experiment = NimbusExperiment.objects.get(id=experiment_id)
-        experiment.results_data = get_experiment_data(experiment)
+        old_results_data = experiment.results_data
+        new_results_data = get_experiment_data(experiment)
+        experiment.results_data = new_results_data
         experiment.save()
```

Comment:
This could probably go inside the `if` block, though I imagine the performance difference is negligible..